packages/directives/__tests__/trap-focus.spec.ts

```diff
@@ -0,0 +1,156 @@
+      (wrapper.element as ITrapFocusElement)[FOCUSABLE_CHILDREN].length,
+    ).toBe(1)
+    expect(
+      (wrapper.element as ITrapFocusElement)[TRAP_FOCUS_HANDLER].length,
+    ).toBeDefined()
+  })
+
+  test('should not fetch disabled element', () => {
+    wrapper = _mount(`
+      <div v-trap-focus>
+        <button />
+        <button disabled />
+        <a href="test" />
+        <a />
+        <input />
+        <input disabled />
+        <input type="hidden" />
+        <input type="file" />
+        <div tabindex="-1" />
+        <select />
+        <textarea />
+      </div>
+    `)
+    expect(
+      (wrapper.element as ITrapFocusElement)[FOCUSABLE_CHILDREN].length,
+    ).toBe(5)
+  })
+
+  test('should trap keyboard.tab event', async () => {
+    wrapper = _mount(`
+    <div v-trap-focus>
+      <button class="button-1" />
+      <button class="button-2" />
+      <button class="button-3" />
+    </div>
+  `)
+
+    expect(document.activeElement).toBe(document.body)
+    await wrapper.find('.button-1').trigger('keydown', {
+      code: 'Tab',
+      shiftKey: true,
+    })
+
+    expect(document.activeElement).toBe(wrapper.find('.button-3').element)
+    await wrapper.find('.button-3').trigger('keydown', {
+      code: 'Tab',
+    })
+
+    expect(document.activeElement).toBe(wrapper.find('.button-1').element)
+    // the current active element is .button-1
+    await wrapper.find('.button-1').trigger('keydown', {
+      code: 'Tab',
+    })
+
+    expect(document.activeElement).toBe(wrapper.find('.button-2').element)
+
+    // now the active element became .button-2, this time we navigate back
+    await wrapper.find('.button-2').trigger('keydown', {
+      code: 'Tab',
+      shiftKey: true,
+    })
+    expect(document.activeElement).toBe(wrapper.find('.button-1').element)
+  })
+
+  test('should prevent tab event when there is only one element', async () => {
+    wrapper = _mount(`
+      <div v-trap-focus>
+        <button />
+      </div>
+    `)
+    expect(document.activeElement).toBe(document.body)
+    await wrapper.find('button').trigger('keydown', {
+      code: 'Tab',
+    })
+    expect(document.activeElement).toBe(document.body)
+  })
+
+  test('should update focusable list when children changes', async () => {
+    wrapper = mount(
+      {
+        props: {
+          show: {
+            type: Boolean,
+            default: false,
+          },
+        },
+        template: `
+      <div v-trap-focus>
+        <button />
+        <button v-if="show" />
+      </div>
+    `,
+      },
+      {
+        global: {
+          directives: {
+            TrapFocus,
+          },
+        },
+      },
+    )
+    const initialElements = wrapper.element[FOCUSABLE_CHILDREN]
+    expect(initialElements.length).toBe(1)
+
+    await wrapper.setProps({
+      show: true,
+    })
+
+    expect(wrapper.element[FOCUSABLE_CHILDREN].length).toBe(2)
+    expect(wrapper.element[FOCUSABLE_CHILDREN]).not.toEqual(initialElements)
```

Comment:
Always return `false` since their lengths are different, do u mean `toBe`？

Reply:
I think you are right, the two test cases are redundant, I should only keep one of them, since the length of the focusable children has already proven `before` and `after` is not the same